docs/defradb/guides/peer-to-peer.md

```diff
@@ -2,143 +2,275 @@
+### PubSub System Overview
 
-One major difference between active and passive networks is that an active network can focus on both collections and individual documents, while a passive network is only focused on individual documents. Active networks operate over a direct, point-to-point connection and allow you to select an entire collection to replicate to another node. For example, if you have a collection of books and specify a target node for active replication, the entire collection will be replicated to that node, including any updates to individual books. However, it is also possible to replicate granularly by selecting specific books within the collection for replication. Passive networks, on the other hand, are only concerned with replicating individual documents.
+- **Publishers** send messages without targeting specific receivers.
+- **Subscribers** express interest in specific types of messages, regardless of who sends them.
+- This model allows for:
+  - Dynamic network topology
+  - Enhanced scalability
+- Nodes in the network can:
+  - **Publish** messages to specific topics
+  - **Subscribe** to topics to receive relevant messages
 
+### Passive Replication Using Gossip Protocol
+
+- When a node **publishes a message**:
+  - It is **broadcasted** to all nodes in the network.
+  - Nodes **re-broadcast** and **gossip** the message via multiple hops.
+  - This behavior is governed by the **Gossip protocol**, ensuring wide dissemination.
+- **Per-document topics**:
+  - Each document is assigned its own topic.
+  - Nodes can subscribe to specific document topics to receive updates.
+- **Hot vs. Cold documents**:
+  - Frequently accessed or updated documents (“hot documents”) keep open connections for timely syncing.
+  - Infrequently accessed ones (“cold documents”) are easily resynced by querying updates as needed.
+
+### Focus of Passive Replication
+
+- Designed for **individual document-level synchronization**
+- Ideal for:
+  - High-demand documents
+  - Environments with dynamic collaboration
+
+### Comparison with Active Replication
+
+- **Active Replication**:
+  - Works over **direct, point-to-point connections**
+  - Supports:
+    - Entire **collection-level replication**
+    - **Granular replication** of selected documents within a collection
+- **Passive Replication**:
+  - Limited to **individual document-level** replication
+  - Operates via the **PubSub network** using the **Gossip protocol**
+
+### Example Use Case
+
+- If you have a **collection of books**:
+  - **Active replication** can replicate the entire collection or just selected books to a specific node.
+  - **Passive replication** focuses only on syncing individual books that nodes are subscribed to.
 
 ## Concrete Features of P2P in DefraDB
```

Comment:
todo: lets remove the "Concrete" term here, its useless (I know this is from the previous state of the document)